runelite-client/src/main/java/net/runelite/client/game/loot/events/NpcLootReceived.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018, TheStonedTurtle <https://github.com/TheStonedTurtle>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.game.loot.events;
+
+import lombok.Data;
+import lombok.Getter;
+import net.runelite.api.NPC;
+import net.runelite.api.NPCComposition;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.game.loot.data.ItemStack;
+import java.util.List;
+
+/**
+ * An event where you killed an {@link NPC} and received loot
+ */
+@Data
+public class NpcLootReceived
+{
+	@Getter
+	private final int npcId;
```

Comment:
i would probably replace the npcId and npccomposition with just the NPC

Reply:
Actor compositions are set to null if they despawn. This would effectively mean you don't have access to their id or name anymore.

Reply:
it worked when i tested it changed it to pass the npc and still could get the name and id

Reply:
Huh. I'm pretty sure originally @TheStonedTurtle wanted to have the implementation you suggested originally but it turned out not to be possible due to what I described.

Reply:
I just tested this, again, passing just the NPC and this is the result:
https://i.imgur.com/5Wobo73.png
And here's the subscribe event
https://i.imgur.com/wq0UriT.png

Reply:
yea i retried it but came to the same result but im sure i had it working earlier then again i have been awake like 20+ hours i will continue messing around and see if i can figure out how i got it to work if i even did